src/components/common/needsShutdown.jsx

```diff
@@ -121,6 +125,10 @@ export function getDevicesRequiringShutdown(vm) {
     if (needsShutdownSpice(vm))
         devices.push(_("SPICE"));
 
+    // TPM
+    if (needsShutdownTpm(vm))
+        devices.push(_("TPM"));
```

Comment:
This added line is not executed by any test.

Reply:
Right you are, coverage tests. I added a check for adding a TPM to a running VM, and test that "needs restart" indicator. This also uncovered a bug: The "Add TPM" action does not go away in this state, i.e. TPM got added to the inactive XML, but not yet the running one.

I fixed that bug and added tests for the "needs shutdown" and "add TPM while running" cases.